docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+1. Add an ingredient into Kitchen Helper.
+   1. Prerequisites: List all the ingredient using the `listingredient all` command. 
+   2. Test case: 'addingredient /n beef /c meat /q 3 /p 20.20 /e 03/03/2020'    
+   Expected: Entry can be seen using `listingredient all` command.
+    
+#### F.3. Search for ingredient
+1. Search for ingredients in Kitchen Helper.
+   1. Prerequisites: The ingredient list should not be empty.
+   2. Test case: `searchingredient beef`  
+   Expected: Ingredient entries that have the keyword matching `beef` names are listed.
+   3. Test case: 'searchingredient meat'  
+   Expected: Ingredient entries that have the keyword matching `meat` category are listed.
+   4. Test case: `searchingredient 02/02/2020`  
+   Expected: Ingredient entries that have the keyword matching `02/02/2020` date are listed.
+   5. Test case: `searchingredient $20`  
+   Expected: Ingredient entries that have the keyword matching `$20` price are listed.
+   
+#### F.4. Search for recipe
+1. Search for similar recipe in Kitchen Helper.
+   1. Prerequisites: The recipe list should not be empty.
+   2. Test case: 'searchrecipe chicken' 
+   Expected: Recipe's name entries that have the keyword matching `chicken' are listed. 
+   
+#### F.5. Search for chore
+1. Search for chores in Kitchen Helper.
+   1. Prerequisites: The chore list should not be empty.
+   2. Test case: `searchchore groceries`  
+   Expected: Chore entries that have the keyword matching `groceries` description are listed.
+   3. Test case: `searchchore Tuesday`  
+   Expected: Chore entries that have the keyword matching `Tuesday` as a string are listed.  
+   
+#### F.6. Saving data
 
-* *glossary item* - Definition
+1. Load ingredient data into Kitchen Helper.
+   1. Prerequisites: The ingredient list save file should not be empty. 
+   1. Expected: Previously stored ingredient data can be seen using `listingredient all` command.
 
-## Instructions for Manual Testing
+1. Load recipe data into Kitchen Helper.
+   1. Prerequisites: The recipe list save file should not be empty. 
+   1. Expected: Previously stored recipe data can be seen using `listrecipe all` command.
+ 
+1. Load chore data into Kitchen Helper.
+   1. Prerequisites: The chore list save file should not be empty. 
+   1. Expected: Previously stored chore data can be seen using `listchore all` command.
+ 
+If any of the save files are empty, the user can choose to populate the files with their own user commands or alternatively, use any of the test cases below:
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+1. `addrecipe /n Chicken Salad /i Chicken Breast:2:meat, Lettuce:4:vegetable`
+2. `addingredient /n Chicken Breast /c meat /q 3 /p 20 /e 18/03/2020`
+3. `addingredient /n kailan /c Vegetable /q 30 /p 30.45 /e 12/03/2020`
+4. `addingredient /n HL Milk /c Dairy /q 3 /p 12.2 /e 14/03/2020`
+5. `addchore buy groceries /by Tuesday 12pm`
```

Comment:
Overall, want to say good job on your developer guide, it is very detailed and provides lot of info. However, this might make it less readable in some cases, so do consider how to change your diagrams if needed. Good job!!!